app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/eventRepository/remote/eventApi/eventApiImpls/EventApiImpl.kt

```diff
@@ -66,43 +87,94 @@ class EventApiImpl @Inject constructor(
-//                                "photos",
-//                                "photo$index",
-//                                body = event.photos[0].toRequestBody()  // todo Convert URI to ByteArray
-//                        )
-//                }.also {
-//                }
+                photos = event.photos.mapIndexed { index, photo ->
+
+                    val bytes = getBytesFromUri(context, photo.uri)?.also { bytes ->
+                        val size = bytes.size
+                        if (size > MAX_IMAGE_SIZE) {
+                            throw NetworkErrorException("Photo too large")
+                        }
+                    } ?: throw Exception("Photo not found")
+
+                    MultipartBody.Part
+                        .createFormData(
+                            "photo$index",
+                            getImageFilenameForServer(context, index, photo.uri, photo.id),
+                            body = bytes.toRequestBody()
+                        )
+                }
             )
             if (response.isSuccessful) {
                 val responseBody = response.body()
-                responseBody ?: throw Exception("Response body is null")
+                return responseBody ?: throw Exception("Response body is null")
             } else {
-                throw Exception("Error updating event: ${response.errorBody()}")
+                throw Exception("Error updating event: ${getErrorBodyMessage(response.errorBody()?.string())}")
             }
+        } catch (e: NetworkErrorException) {
+            throw Exception("Network Error updating event: ${e.localizedMessage}")
+        } catch (e: Exception) {
+            throw Exception("${e.message}")
+        }
+    }
+
+    override suspend fun deleteEvent(eventId: EventId): Boolean {
+        return try {
+            val response = taskyApi.deleteEvent(eventId)
+            response.isSuccessful
         } catch (e: Exception) {
-            throw Exception("Error updating event: ${e.message}")
+            false
         }
     }
+
+    private fun getImageFilenameForServer(
+        context: Context,
+        index: Int,
+        uri: Uri,
+        id: UuidStr
+    ): String {
+        return "photo$index-$id." +
+                (InputStreamRequestBody
+                    .getFileName(context, uri)
+                    ?.split(".")
+                    ?.last()
+                    ?: "jpg")
+    }
+
+    // Will automatically recompress to lower quality if the image is too large
+    private fun getBytesFromUri(context: Context, uri: Uri): ByteArray?  {
+        var bytes = context.contentResolver
+            .openInputStream(uri)
+            .use {
+                it?.readBytes()
+            }
+
+        val imageSize = context.contentResolver.openFileDescriptor(uri, "r")
+            .use {
+                it?.statSize
+            }
+
+        imageSize?.also { size ->
+            if (size > 1000000) {
```

Comment:
Should use your `MAX_IMAGE_SIZE` constant I guess

Reply:
OK, will use the constant.